### PR TITLE
[VFX] Skinned Mesh Sampling (to 2021.1)

### PR DESCRIPTION
**Pull request by PaulDemeulenaere**
Actually a backport of internal [PR](https://github.cds.internal.unity3d.com/unity/vfx-graphics/pull/30), which is landing (or landed) with vfx/staging at https://github.com/Unity-Technologies/Graphics/pull/3197

---
### Purpose of this PR
- Add Skinned Mesh Renderer Sampling
- Add triangle sampling for Mesh Sampling
- Add support of color which are stored as float

**Node addition**
- Skinned Mesh Renderer Index & Vertex Count
image

- New variant of Sample Mesh, "Sample Skinned Mesh" :
image

Both now supports the placement option :
- Vertex : 
image

- Edge :
image
_The index correspond to the starting index of the edge, the X is a linear interpolation value between starting index and next index (triangle are automatically looped)._

- Surface :
image

And supports two coordinate mode :
- Uniform which is using the surface placement described by Eric Heitz. I could readd the classical square root placement, I think we have to choose between those two, in green, Heitz, in red, square root approach, here, we chose green :
_coord
- Barycentric : Raw barycentric coordinate, x, y are exposed and z is computed from input value (z = 1 - x -y )

- New variant of Position Mesh, "Sample Skinned Mesh" :
Vertex : if random, pick a vertices among all vertexCount else, same input than Sample Mesh
Edge : if random, pick an edge and a random value [0, 1[ for interpolation Not uniform else, same input than Sample Mesh
Surface : if random, pick an triangle (indexCount / 3u) and pick two random value [0, 1[ to sample in uniform triangle uniform from triangle point of view but not uniform from mesh, else, sample input than Sample Mesh
_sample_mesh

---
### Testing status
- See [QA testing document](QA testing document : https://docs.google.com/document/d/1VbIc5pMkgFX2Ui4-LWrZ7uQ4EKlXlFxWWW4ufcex70I/edit)
- New graphic test
_graphic_test_skinning

---
### Comments to reviewers
This internal PR : https://github.cds.internal.unity3d.com/unity/ScriptableRenderPipelinePrivate/pull/78 should be synchronously merged.

### CODE REVIEW

File: com.unity.visualeffectgraph/CHANGELOG.md [hunk 1]
```diff
@@ -8,9 +8,14 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
...
 - Added new setting to output nodes to exclude from TAA
 - New Sample Point cache & Sample Attribute map operators
+- Sample vertices of a transformed skinned mesh with Position (Skinned Mesh) and Sample Skinned Mesh operator.
+- Placement option (Vertex, Edge, Surface) in Sample Mesh & Skinned Mesh, allows triangle sampling.
 
 ### Changed
 - Changed the "Edit" button so it becomes "New" when no asset is set on a Visual Effect component, in order to save a new visual effect graph asset.
+- Serialization issue with VFXSpawnerCallbacks
+- Unexpected exception while trying to display capacity warning [Case 1294180](https://issuetracker.unity3d.com/product/unity/issues/guid/1294180/)
+- Exclude Operator, Context, Block and Subgraph from Preset [Case 1232309](https://issuetracker.unity3d.com/product/unity/issues/guid/1232309/)
```
Comment (julienf-unity) [inline com.unity.visualeffectgraph/CHANGELOG.md:18]: There seem to be  unrelated entries here

**Review by julienf-unity — CHANGES_REQUESTED** [review]

File: com.unity.visualeffectgraph/CHANGELOG.md (hunk 1 above)
Comment (PaulDemeulenaere) [reply com.unity.visualeffectgraph/CHANGELOG.md:18]: Ah yes, sorry, probably a wrong merge, I will double check from the initial change at : https://github.cds.internal.unity3d.com/unity/vfx-graphics/pull/30

Comment (PaulDemeulenaere) [reply com.unity.visualeffectgraph/CHANGELOG.md:18]: Fixed at 1d47db5057a62fe86551d6464381258b243c94f2

**Comment by PaulDemeulenaere** [comment]
All nightly VFX job are green at https://github.com/Unity-Technologies/Graphics/pull/3451/commits/688e280334f91536b093d616a82a8bde4a3591e7, there is no need to relaunch Yamato, I only updated the changelog (so far).

**Review by julienf-unity — APPROVED** [review]

**Pull request merged**